### PR TITLE
feat: Add create Zapp workflow APIs

### LINKED ISSUE #33: Make APIs to create Zapp
## Type of PR

- [x] Feature Request
- [ ] Report Bug
- [ ] Documentation Update
- [ ] Other

## Description
Create APIs that are required to make a zap, Follow workflow for that. 

## Check
- [x] I have followed issue creation guidelines
- [x] I have attached proper label with the issue
- [x] The issue is assigned to a assignee

### PR DESCRIPTION
**Pull request by harish-aka-shivi**
This branch is branched off another branch ( #43 ) which is yet to be merged, so there are two commits.
That is the reason there are so many file changes.

This PR would add the following APIs at least:
- create Zapp
- delete Zapp
- or, any other helper API

It also fixes some minor typos.
Fixes #33 #37 #36 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Unit Test
- [ ] Integration Test

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: backend/src/lib/requestPromisified/index.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import request from 'request';
+
+// this function create the normal request object in a promise.
+const requestPromisified = ({
```
Comment (nimish-gupta) [inline backend/src/lib/requestPromisified/index.js]: this is a duplicate code from file ` backend/src/lib/requestPromisified.js`

File: backend/src/__tests__/zappRouter.test.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    expect(resZappDel.statusCode).toEqual(200);
+  });
+
+  afterAll(async () => {
+    const resDelete = await request(app)
+      .delete('/users/me')
+      .set('Authorization', `Bearer ${authToken}`);
+    // done();
+    app.close();
+    console.log(resDelete.body);
```
Comment (nimish-gupta) [inline backend/src/__tests__/zappRouter.test.js]: console.log ?

File: backend/src/lib/requestPromisified/index.js (hunk 1 above)
Comment (harish-aka-shivi) [reply backend/src/lib/requestPromisified/index.js]: I will remove it and just keep the folder instead.

File: backend/src/__tests__/zappRouter.test.js (hunk 2 above)
Comment (harish-aka-shivi) [reply backend/src/__tests__/zappRouter.test.js]: Will remove it.

**Pull request merged**